src/lib.rs

```diff
@@ -114,6 +114,38 @@ pub enum SecParam {
 }
 
 impl SecParam {
+    /// Convert an integer input to the corresponding security parameter.
+    /// # Arguments
+    ///
+    /// * `value` - An integer representing the desired security parameter.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(SecParam)` - If the conversion is successful, returns the corresponding `SecParam`.
+    /// * `Err(OperationError)` - If the given `value` does not correspond to any supported security parameter.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use capycrypt::SecParam;
+    /// use capycrypt::OperationError;
+    ///
+    /// assert_eq!(SecParam::from_int(224).unwrap(), SecParam::D224);
+    /// assert_eq!(SecParam::from_int(256).unwrap(), SecParam::D256);
+    /// assert_eq!(SecParam::from_int(384).unwrap(), SecParam::D384);
+    /// assert_eq!(SecParam::from_int(512).unwrap(), SecParam::D512);
+    /// assert_eq!(SecParam::from_int(1024), Err(OperationError::UnsupportedSecurityParameter))
+    /// ```
```

Comment:
these doc tests can be really hard to maintain if we ever need to change the security parameters. I would recommend removing this section and simply stating the supported security params as simple comments instead